controllers/v1/programs.js

```diff
@@ -3,13 +3,593 @@
+    * {
+        "message": "Successfully removed entities in program scope",
+        "status": 200
+      }
+    */
+
+    /**
+   * Remove entities in program scope
+   * @method
+   * @name removeEntitiesInScope
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - program id.
+   * @param {Array} req.body.entities - Entities to be added.
+   * @returns {Array} Program scope roles.
+   */
+
+  async removeEntitiesInScope(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let programUpdated = await programsHelper.removeEntitiesInScope(
+          req.params._id,
+          req.body.entities
+        );
+    
+        return resolve(programUpdated);
+
+      } catch (error) {
+        return reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        });
+      }
+    });
+  }
+
+  /**
+  * @api {post} /project/v1/programs/list?page=:page&limit=:limit&search=:search
+  * @apiVersion 1.0.0
+  * @apiName 
+  * @apiGroup Programs
+  * @apiParamExample {json} Request-Body:
+  * {
+      "entities" : ["5f33c3d85f637784791cd830"]
+    }
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /project/v1/programs/list?page=1&limit=10&search=PROGRAM01
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  {
+      "message": "Program lists fetched successfully",
+      "status": 200,
+      "result": {
+          "data": [
+              {
+                  "_id": "5beaaaa6af0065f0e0a10605",
+                  "externalId": "APPLE-ASSESSMENT-PROGRAM",
+                  "description": "Apple Program 2018",
+                  "isAPrivateProgram": false
+              }
+          ],
+          "count": 1
+      }
+  }
+  */
+
+   /**
+   * List programs.
+   * @method
+   * @name list
+   * @param {Object} req - Requested data.
+   * @param {Array} req.query.page - Page number.
+   * @param {Array} req.query.limit - Page Limit.
+   * @param {Array} req.query.search - Search text.
+   * @returns {JSON} List programs data.
+  */
+
+ async list(req) {
+  return new Promise(async (resolve, reject) => {
+    try {
+
+      let listOfPrograms = await programsHelper.list(
+        req.pageNo,
+        req.pageSize,
+        req.searchText
+      );
+
+      return resolve(listOfPrograms);
+
+    } catch (error) {
+      return reject({
+        status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+        message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+        errorObject: error
+      });
+    }
+  });
+}
+
+
+
+
+
+ /**
+    * @api {get} /project/v1/programs/join/:programId 
+    * @apiVersion 1.0.0
+    * @apiName Program Join
+    * @apiGroup Programs
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
```

Comment:
@Prajwal17Tunerlabs  add sample  request and response

Reply:
api doc updated